src/VisualStudio/Core/Def/Implementation/SemanticClassificationCache/VisualStudioSemanticClassificationCacheService.cs

```diff
@@ -52,18 +52,16 @@ internal class VisualStudioSemanticClassificationCacheService
                 return default;
             }
 
-            var classifiedSpans = await client.RunRemoteAsync<SerializableClassifiedSpans>(
-                WellKnownServiceHubService.CodeAnalysis,
-                nameof(IRemoteSemanticClassificationCacheService.GetCachedSemanticClassificationsAsync),
-                solution: null,
-                arguments: new object[] { documentKey.Dehydrate(), textSpan, checksum },
+            var classifiedSpans = await client.TryInvokeAsync<IRemoteSemanticClassificationCacheService, SerializableClassifiedSpans?>(
+                (service, cancellationToken) => service.GetCachedSemanticClassificationsAsync(documentKey.Dehydrate(), textSpan, checksum, cancellationToken),
```

Comment:
how does SB generally handle argument marshalling?  same way as today ?  (i.e. json rpc)?  pluggable?

Reply:
The same as today. We can use JSON or MessagePack. I have a follow up PR to switch to MessagePack.